presidio_evaluator/data_generator/presidio_data_generator.py

```diff
@@ -295,48 +292,119 @@ def name_gendered(row):
+                                         organization="ORGANIZATION",
+                                         name_female="PERSON",
+                                         address="STREET_ADDRESS",
+                                         country="GPE",
+                                         state="GPE",
+                                         credit_card_number="CREDIT_CARD",
+                                         city="GPE",
+                                         street_name="STREET_ADDRESS",
+                                         building_number="STREET_ADDRESS",
+                                         name="PERSON",
+                                         iban="IBAN_CODE",
+                                         last_name="PERSON",
+                                         last_name_male="PERSON",
+                                         last_name_female="PERSON",
+                                         first_name="PERSON",
+                                         first_name_male="PERSON",
+                                         first_name_female="PERSON",
+                                         phone_number="PHONE_NUMBER",
+                                         url="DOMAIN_NAME",
+                                         ssn="US_SSN",
+                                         email="EMAIL_ADDRESS",
+                                         date_time="DATE_TIME",
+                                         date_of_birth="DATE_TIME",
+                                         day_of_week="DATE_TIME",
+                                         year="DATE_TIME",
+                                         name_male="PERSON",
+                                         prefix_male="TITLE",
+                                         prefix_female="TITLE",
+                                         prefix="TITLE",
+                                         nationality="NRP",
+                                         nation_woman="NRP",
+                                         nation_man="NRP",
+                                         nation_plural="NRP",
+                                         first_name_nonbinary="PERSON",
+                                         postcode="STREET_ADDRESS",
+                                         secondary_address="STREET_ADDRESS",
+                                         job="TITLE",
+                                         zipcode="ZIP_CODE",
+                                         state_abbr="GPE",
+                                         age="AGE")
+
+    def __init__(self,
+                 locale: str,
+                 lower_case_ratio: float,
+                 additional_entity_providers: List[BaseProvider] = None,
+                 additional_sentence_templates: List[str] = None,
+                 random_seed: SeedType = None):
+        presidio_templates_file_path = raw_data_dir / "templates.txt"
```

Comment:
With the current API, a user can load the sentences in before constructing this class, and pass them in as a list of strings to the `additional_sentence_templates` parameter e.g.
```python
my_custom_templates = PresidioDataGenerator.read_template_file('~/my_templates.txt')
my_custom_templates.extend(PresidioDataGenerator.read_template_file('~/other_dir/more_templates.txt'))

record_generator = PresidioFakeRecordGenerator(locale='en', lower_case_ratio=0, additional_sentence_templates=my_custom_templates)
```

Reply:
I've been viewing this class as a high-level API which, by default, will use all the power/functionality provided by this library. Just as the code that already existed in the main function and the notebook was there as what a layman would run
Primarily to target the ML Engineer persona who wishes to evaluate the configurable models for generic PII detection, rather than the Data Scientist persona who wants to build their model / drop lower level into configuring templates/providers
If they wish to strip back by using fewer templates or fewer providers, that's when they drop to the lower-level API

Though it would be nice to be able to easily choose to extend or override, at the moment, you can only override
One option would be a flag parameter for extending or overriding, but I think that makes a clunky API
Or another option is defaulting to the existing templates and providers, and override these with those provided instead of extending. I think it makes sense to provide this default for ease of use, and it doesn't make sense to default to nothing as then the constructed object is unusable. If overriding instead of extending, then just need to make it easy to load all existing templates/providers prior to constructing the class to accommodate that use case

Reply:
I updated the API to override instead of extending, to support not wanting to use the inbuilt templates/providers
https://github.com/microsoft/presidio-research/pull/50/commits/407718d1827751a0f2116a633dc2e00b4cf5fbfe